stl/CMakeLists.txt

```diff
@@ -458,11 +458,11 @@ function(add_stl_dlls D_SUFFIX THIS_CONFIG_DEFINITIONS THIS_CONFIG_COMPILE_OPTIO
     target_compile_options(msvcp${D_SUFFIX}_eha_objects PRIVATE "${THIS_CONFIG_COMPILE_OPTIONS};${GL_FLAG};/EHa")
 
     add_library(msvcp${D_SUFFIX} SHARED)
-    target_link_libraries(msvcp${D_SUFFIX} PRIVATE msvcp${D_SUFFIX}_eha_objects msvcp${D_SUFFIX}_objects msvcp${D_SUFFIX}_init_objects "${TOOLSET_LIB}/vcruntime${D_SUFFIX}.lib" "${TOOLSET_LIB}/msvcrt${D_SUFFIX}.lib" "ucrt${D_SUFFIX}.lib")
+    target_link_libraries(msvcp${D_SUFFIX} PRIVATE msvcp${D_SUFFIX}_eha_objects msvcp${D_SUFFIX}_objects msvcp${D_SUFFIX}_init_objects "${TOOLSET_LIB}/vcruntime${D_SUFFIX}.lib" "${TOOLSET_LIB}/msvcrt${D_SUFFIX}.lib" "ucrt${D_SUFFIX}.lib" "ole32.lib" "delayimp.lib")

```

Comment:
we can't add dependencies to our existing dlls. Ever.

If a program linked with the latest stl is loaded into a process that uses an older one, these dlls will not be loaded. 

Besides this we can't use delay loading because it's a massive footgun.

Reply:
Why not? If a process loads a dll with the ole32.dll dependency, the loader will set to it that it's loaded as well (if not already).

Reply:
Edit: we actually can, since the dependent code travels along for the ride.

Reply:
so, to clarify - good with the ole32.dll dependency, just not delay loading?

Reply:
According to my (imperfect) understanding:
* We should be able to add an ole32.dll dependency as long as it is present on all of the platforms we care about (we no longer care about Vista / Server 2008 classic)
* Delay-loading is perilous due to loader lock issues (which I forgot about until Charlie and Casey reminded me), static dependencies aren't problematic in that way

Reply:
Agreed - will remove delayload from both PRs